lib/src/runner/runtime_utils.dart

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) 2015, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// JS runtime files utilities used by dartdevrun.
+library dev_compiler.src.runner.runtime_utils;
+
+import 'dart:async';
+import 'dart:io';
+
+import 'file_utils.dart' show listJsFiles;
+import 'package:dev_compiler/src/options.dart';
+import 'package:path/path.dart';
+
+/// In node.js / io.js, these modules need to be aliased globally
+/// (e.g. `var foo = require('./path/to/foo.js')`).
+const _ALIASED_RUNTIME_FILES = const {
+  'dart_library.js': 'dart_library',
+  'dart_utils.js': 'dart_utils',
+};
+
+/// Runtime files that should appear first, because they define fundamental
+/// facilities (module system, etc) or they do crucial sanity checks.
+final _PRIORITARY_RUNTIME_FILES = []
+  ..addAll(_ALIASED_RUNTIME_FILES.keys)
+  ..addAll([
+    'harmony_feature_check.js',
+  ]);
+
+final _BLACKLISTED_RUNTIME_FILES = new Set<String>()
+  ..addAll([
+    'messages_widget.js',
+    // TODO(ochafik): Is this one mean to run?
+    'run.js',
+  ]);
+
+/// If [path] is a runtime file with an alias, returns that alias, otherwise
+/// returns null.
+String getRuntimeFileAlias(CompilerOptions options, String path) =>
+    path.startsWith(options.runtimeDir)
+    ? _ALIASED_RUNTIME_FILES[basename(path)] : null;
+
+Future<List<File>> listOutputFiles(CompilerOptions options) async =>
+    listJsFiles(new Directory(options.codegenOptions.outputDir));
+
+/// Lists runtime files from the runtime dir
+Future<List<File>> listRuntimeFiles(CompilerOptions options) async {
+  getPriorityIndex(file) {
+    int i = _PRIORITARY_RUNTIME_FILES.indexOf(basename(file.path));
+    return i < 0 ? _PRIORITARY_RUNTIME_FILES.length : i;
+  }
+  List<File> files = (await listJsFiles(new Directory(options.runtimeDir)))
+      .where((f) => !_BLACKLISTED_RUNTIME_FILES.contains(basename(f.path)))
+      .toList();
+  return files..sort((File a, File b) {
+    int pa = getPriorityIndex(a),
+        pb = getPriorityIndex(b);
+    return pa != pb ? (pa - pb) : a.path.compareTo(b.path);
+  });
+}
+
+List<String> _turnFilesToUris(List<String> args) =>
+    args.map((arg) {
+      var file = new File(arg);
+      return file.existsSync() ? "file:" + file.absolute.path : arg;
+    }).toList();
+
+String getMainModuleName(CompilerOptions options) {
+  // This is an ugly hack to guess the path to the output entry point .js.
+  // TODO(ochafik): Find / create a clean way to get the output .js file(s).
```

Comment:
I think you want getModulePath and/or getOutputPath
https://github.com/dart-lang/dev_compiler/blob/0b0f07faea680f4f8b3ffaa8854f82be4febc960/lib/src/compiler.dart#L309